### PR TITLE
chore: add cli command to fetch group sync settings as json

### LINKED ISSUE #14687: Implement Group and Role Sync CLI
We need to create a CLI where users can handle Group and Role sync CRUD operations. For now, we are going to handle update and read and will defer other operations until after GA. 

**Requirements**
- [ ] accepts a JSON file as an input
- [ ] reads JSON back to user

API Endpoint to read: https://github.com/coder/coder/blob/812a78637229eb33f344024d03ee4a4eb7fd32aa/enterprise/coderd/idpsync.go#L14-L22

**Note:** Experience will intentionally be minimal for GA. UX can be improved after GA.

### PR DESCRIPTION
**Pull request by Emyrk**
CLI commands to read and update role + group sync runtime settings. All settings are in json and take in input via `stdin`. Currently the cli is intentionally a bit obtuse in it's UX.

This is an MVP for setting the runtime values. The CLI and UI experience will be improved after GA. 

### Groupsync

```shell
$ coder organization  settings show groupsync --org=rabbit                                                                                               
{
        "field": "",
        "mapping": null,
        "regex_filter": null,
        "auto_create_missing_groups": false
}
```

And setting

```shell
$ coder organization  settings set groupsync --org=rabbit < input.json
```

### Rolesync

```shell
$ coder organization  settings show rolesync --org=rabbit
{
        "field": "",
        "mapping": null
}
```

And setting

```shell
$ coder organization  settings set rolesync --org=rabbit < input.json
```

## Improved error message:

Probably should have thrown this in another PR. In testing, I improved the error message for my own convenience.

```
error: Must select an organization with --org=<org_name>. Choose from "coder", "rabbit"
```

Closes https://github.com/coder/coder/issues/14687

### CODE REVIEW

File: cli/organizationsettings.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+func (r *RootCmd) organizationSettings(orgContext *OrganizationContext) *serpent.Command {
+	cmd := &serpent.Command{
+		Use:     "settings",
+		Short:   "Manage organization settings.",
+		Aliases: []string{"setting"},
+		Handler: func(inv *serpent.Invocation) error {
+			return inv.Command.HelpHandler(inv)
+		},
+		Hidden: true,
```
Comment (f0ssel) [inline cli/organizationsettings.go:24]: Should this be hidden since this will be in GA?

File: cli/organizationsettings.go [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+			}
+
+			// Read in the json
+			inputData, err := io.ReadAll(inv.Stdin)
+			if err != nil {
+				return xerrors.Errorf("reading stdin: %w", err)
+			}
+
+			var setting any
+			switch strings.ToLower(inv.Args[0]) {
```
Comment (f0ssel) [inline cli/organizationsettings.go:63]: This feels like we are reinventing cobra sub commands. Could we compose these as true sub commands?

File: cli/organizationsettings.go [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+		),
+		Handler: func(inv *serpent.Invocation) error {
+			ctx := inv.Context()
+			org, err := orgContext.Selected(inv, client)
+			if err != nil {
+				return err
+			}
+
+			var setting any
+			switch strings.ToLower(inv.Args[0]) {
```
Comment (f0ssel) [inline cli/organizationsettings.go:129]: Same sub command request

File: cli/organizationsettings.go (hunk 1 above)
Comment (Emyrk) [reply cli/organizationsettings.go:24]: Such a habit, no! Will remove.

File: cli/organizationsettings.go (hunk 2 above)
Comment (Emyrk) [reply cli/organizationsettings.go]: Fixed:
```
USAGE:
  coder organizations settings set

  Update specified organization setting.

    - Update group sync settings.:
  
        $ coder organization settings set groupsync < input.json 

SUBCOMMANDS:
    group-sync    Group sync settings to sync groups from an IdP.
    role-sync     Role sync settings to sync organization roles from an IdP.

———
Run `coder --help` for a list of global options.
```

**Review by f0ssel — APPROVED** [review]

**Pull request merged**